comps/dataprep/opensearch/README.md

```diff
@@ -0,0 +1,283 @@
+
+```bash
+curl localhost:$your_port/embed \
+    -X POST \
+    -d '{"inputs":"What is Deep Learning?"}' \
+    -H 'Content-Type: application/json'
+```
+
+After checking that it works, set up environment variables.
+
+```bash
+export TEI_ENDPOINT="http://localhost:$your_port"
+```
+
+### 1.4 Start Document Preparation Microservice for OpenSearch with Python Script
+
+Start document preparation microservice for OpenSearch with below command.
+
+- option 1: Start single-process version (for 1-10 files processing)
+
+```bash
+cd langchain
+python prepare_doc_opensearch.py
+```
+
+<!-- - option 2: Start multi-process version (for >10 files processing)
+
+```bash
+cd langchain_ray
+python prepare_doc_opensearch_on_ray.py
+``` -->
+
+## 🚀2. Start Microservice with Docker (Option 2)
+
+### 2.1 Start OpenSearch Stack Server
+
+Please refer to this [readme](../../vectorstores/opensearch/README.md).
+
+### 2.2 Setup Environment Variables
+
+```bash
+export EMBEDDING_MODEL_ID="BAAI/bge-base-en-v1.5"
+export TEI_ENDPOINT="http://${your_ip}:6006"
+export OPENSEARCH_URL="http://${your_ip}:9200"
+export INDEX_NAME=${your_index_name}
+export HUGGINGFACEHUB_API_TOKEN=${your_hf_api_token}
+```
+
+### 2.3 Build Docker Image
+
+- Build docker image with langchain
+
+- option 1: Start single-process version (for 1-10 files processing)
+
+```bash
+cd ../../
+docker build -t opea/dataprep-opensearch:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/dataprep/opensearch/langchain/Dockerfile .
+```
+
+<!-- - option 2: Start multi-process version (for >10 files processing)
+
+```bash
+cd ../../../
+docker build -t opea/dataprep-on-ray-opensearch:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/dataprep/opensearch/langchain_ray/Dockerfile .
+``` -->
+
+### 2.4 Run Docker with CLI (Option A)
+
+- option 1: Start single-process version (for 1-10 files processing)
+
+```bash
+docker run -d --name="dataprep-opensearch-server" -p 6007:6007 --runtime=runc --ipc=host -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e OPENSEARCH_URL=$OPENSEARCH_URL -e INDEX_NAME=$INDEX_NAME -e TEI_ENDPOINT=$TEI_ENDPOINT -e HUGGINGFACEHUB_API_TOKEN=$HUGGINGFACEHUB_API_TOKEN opea/dataprep-opensearch:latest
+```
+
+<!-- - option 2: Start multi-process version (for >10 files processing)
+
+```bash
+docker run -d --name="dataprep-opensearch-server" -p 6007:6007 --runtime=runc --ipc=host -e http_proxy=$http_proxy -e https_proxy=$https_proxy -e OPENSEARCH_URL=$OPENSEARCH_URL -e INDEX_NAME=$INDEX_NAME -e TEI_ENDPOINT=$TEI_ENDPOINT -e HUGGINGFACEHUB_API_TOKEN=$HUGGINGFACEHUB_API_TOKEN -e TIMEOUT_SECONDS=600 opea/dataprep-on-ray-opensearch:latest
+``` -->
+
+### 2.5 Run with Docker Compose (Option B - deprecated, will move to genAIExample in future)
+
+```bash
+# for langchain
+cd comps/dataprep/opensearch/langchain
+# common command
+docker compose -f docker-compose-dataprep-opensearch.yaml up -d
```

Comment:
Seems like you only use the `docker-compose-dataprep-opensearch.yaml` to start dataprep service. Please remove the `docker-compose-dataprep-only.yaml` file.